zerver/lib/user_status.py

```diff
@@ -45,6 +51,8 @@ def format_user_status(row: RawUserInfoDict) -> UserInfoDict:
         dct["emoji_name"] = emoji_name
         dct["emoji_code"] = emoji_code
         dct["reaction_type"] = reaction_type
+    if row["user_profile_id"] == user_id and scheduled_end_time:
+        dct["scheduled_end_time"] = datetime_to_timestamp(scheduled_end_time)
```

Comment:
I feel like this block wants a comment explaining the conditionals. Something like:

> Scheduled end times are not shared with other users.